### PR TITLE
Organization cover image

### LINKED ISSUE #2068: Add fullpage width background image in organization view
As organization manager I would love to add fullpage width image in the upper part of the organisation view. Now the part is grey color (see attached)
screenshot at helmikuuta 06 13-21-57
. This really would make the page look like "ours".

# Goal

Organization manager can add, edit and delete a cover image. The cover image is displayed in the header of the organization page.Respect the aspect ratio of the image. If needed, hide part of the image.

# Definition of done
To be defined by assignee

# Wireframes
1. In Settings tab for Organization profile, the General section contains a  cover photo uploader. It appears with additional help text
organizationsettings4none

2. On selecting a photo to upload, a preview of the photo appears at the Organization profile header.
User can drag the photo to reposition it. 
This is indicated by a text on transparent panel.
Save and Cancel button appears below the picture to save the picture after repositioning or cancel uploading.
While this view appears, logo picture (if any) and Organization information panes (e.g. Name, Description, Link and Contact info, transparent panel to hold the beforementioned information, social media platform, etc.) will disappear. [e.g. repositioning cover photo in facebook]
organizationsettings4dragcover

3. After Save is clicked, the cover picture appears according to the reposition on Organization profile.
Organization information panes (e.g. Name, Description, Link and Contact info) reappear.
organizationsettings4onlycover

### PR DESCRIPTION
**Pull request by nnn**
Closes #2068

### CODE REVIEW

File: organizations/client/profile/header/header.html [hunk 1]
```diff
@@ -11,6 +11,10 @@
           {{> viewOrganizationLogo organization=organization }}
           <span class="fa fa-camera"></span>
         </div>
+        <button id="organization-cover-file-browse" class="btn btn-default pull-right">
+          <span class="fa fa-camera"></span>
+          Upload cover image
```
Comment (brylie) [inline organizations/client/profile/header/header.html]: Remember i18n

File: organizations/cover/client/upload/uploadButton/uploadButton.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!-- Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
+
+<template name="uploadOrganizationCoverButton">
+  <button class="btn btn-primary fileBrowse">
+    Upload
```
Comment (brylie) [inline organizations/cover/client/upload/uploadButton/uploadButton.html]: Remember i18n

**Comment by brylie** [comment]
@jawidahmadi this issue is now ready for review.

File: organizations/cover/client/upload/upload.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Template.uploadOrganizationCover.onCreated(function () {
+  const instance = this;
+
+  // Subscribe to Organization cover
+  instance.subscribe('organizationCover');
+});
+
+Template.uploadOrganizationCover.helpers({
+  uploadedOrganizationCoverFile () {
+    const organizationCoverFileId = Organizations.findOne().organizationCoverFileId;
```
Comment (brylie) [inline organizations/cover/client/upload/upload.js]: For safety, we should pass the `organization._id` in to the `findOne()`. We have had bugs related to loosely defined `findOne()` queries before, where there could be multiple available documents from active subscriptions.

**Comment by brylie** [comment]
As an aside, I am noticing that file handling in our project takes quite a lot of boilerplate code and workarounds. E.g.

- deleting previous uploaded file
- instantiating resumable uploader in `autorun`
- manually creating file URL (with md5 etc)
- manually creating reference from related entity (organization here) to `fileId`

I am not sure that we can do anything here, but would be really excited to find a clean file uploading package that handles the above situations for us.

**Comment by philippeluickx** [comment]
Maybe we should also include the designer in the review?

**Comment by nnn** [comment]
@brylie looks like a good opportunity for the experiment Monday :)

**Comment by nnn** [comment]
@philippeluickx yep, already included them on Friday, and got their's approval.

**Comment by nnn** [comment]
@jawidahmadi please review.

**Comment by marla-singer** [comment]
@nnn Organization profile looks awesome :+1:

File: organizations/cover/client/upload/upload.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+Template.uploadOrganizationCover.helpers({
+  uploadedOrganizationLogoFile () {
+    const organizationCoverFileId = Organizations.findOne().organizationCoverFileId;
+
+    let organizationCoverFile;
+
+    if (organizationCoverFileId) {
+      // Convert to Mongo ObjectID
```
Comment (marla-singer) [inline organizations/cover/client/upload/upload.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/upload.js [hunk 5]
```diff
@@ -34,3 +34,35 @@ Template.uploadOrganizationCover.helpers({
     return organizationCoverFile;
   },
 });
+
+Template.uploadOrganizationCover.events({
+  'click .delete-organizationCover': function (event, templateInstance) {
+    // Show confirmation dialog to user
```
Comment (marla-singer) [inline organizations/cover/client/upload/upload.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/upload.js [hunk 6]
```diff
@@ -43,7 +43,7 @@ Template.uploadOrganizationCover.events({
   'click .delete-organizationCover': function (event, templateInstance) {
     // Show confirmation dialog to user
```
Comment (marla-singer) [inline organizations/cover/client/upload/upload.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/uploadButton/uploadButton.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
+
+// Meteor packages imports
+import { Template } from 'meteor/templating';
+
+// Collection imports
+import OrganizationCover from '/organizations/cover/collection/collection';
+
+Template.uploadOrganizationCoverButton.onRendered(() => {
+  // Assign resumable browse to element
```
Comment (marla-singer) [inline organizations/cover/client/upload/uploadButton/uploadButton.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/resumable.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+  // TODO: refactor this code, so we can use the arrow-body-style rule
+  // eslint-disable-next-line arrow-body-style
+  OrganizationCover.resumable.on('fileAdded', (file) => OrganizationCover.insert({
+    _id: file.uniqueIdentifier,
+    filename: file.fileName,
+    contentType: file.file.type,
+  }, (err) => {
+    if (err) {
+      // Create & show a message about failed insert
```
Comment (marla-singer) [inline organizations/cover/client/upload/resumable.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/resumable.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+      const message = `${TAPi18n.__('organizationCover_resumable_errorText')} ${err}`;
+      sAlert.error(message);
+      return;
+    }
+
+    // Get organization
+    const organization = Organizations.findOne();
+    // If user doesn't have permission to insert cover
+    if (organization && !organization.currentUserCanManage()) {
+      // Create & show error message about permissions
```
Comment (marla-singer) [inline organizations/cover/client/upload/resumable.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/upload/resumable.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      const message = TAPi18n.__('organizationCover_noPermissions');
+      sAlert.error(message);
+      return;
+    }
+
+    // Accepted extensions for images
+    const acceptedExtensions = ['jpg', 'jpeg', 'png', 'gif'];
+
+    if (fileNameEndsWith(file.file.name, acceptedExtensions)) {
+      // Upload organization cover
```
Comment (marla-singer) [inline organizations/cover/client/upload/resumable.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/cover/client/view.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+Template.viewOrganizationCover.onCreated(function () {
+  const instance = this;
+  // Subscribe to Organization cover
+  instance.subscribe('organizationCover');
+});
+
+Template.viewOrganizationCover.helpers({
+  uploadedOrganizationCoverLink () {
+    // Get organization from template data
```
Comment (marla-singer) [inline organizations/cover/client/view.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

File: organizations/client/profile/header/header.js [hunk 12]
```diff
@@ -11,10 +11,14 @@ import { Modal } from 'meteor/peppelg:bootstrap-3-modal';
 
 // Collection imports
 import OrganizationLogo from '/organizations/logo/collection/collection';
+import OrganizationCover from '/organizations/cover/collection/collection';
 
 Template.organizationProfileHeader.onRendered(function () {
   // Assign resumable browse to element
```
Comment (marla-singer) [inline organizations/client/profile/header/header.js]: @jawidahmadi Your remarks contradict our ESlint rules "padded-blocks"

**Comment by marla-singer** [comment]
@jawidahmadi Actually, we have ESlint rules for regulating spaces and empty lines like "padded-blocks" that reads **"Block must not be padded by blank lines"**. The most part of your request changes contradict this rule

File: organizations/client/profile/header/header.html [hunk 13]
```diff
@@ -4,78 +4,69 @@
...
+            <i class="fa fa-envelope" aria-hidden="true"></i>
+            <a href="mailto:{{ organization.contact.email }}">
+              {{ organization.contact.email }}
+            </a>
+          </li>
+        {{/ if }}
+      </ul>
+      {{# if organization.description }}
+      <p>
+        {{ organization.description }}
```
Comment (marla-singer) [inline organizations/client/profile/header/header.html]: @nnn Make sure nested elements has correct indentation

**Pull request merged**